cyclonedx/model/definition.py

```diff
@@ -15,21 +15,391 @@
+
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, CreId):
+            return hash(other) == hash(self)
+        return False
+
+    def __lt__(self, other: Any) -> bool:
+        if isinstance(other, CreId):
+            return self._id < other._id
+        return NotImplemented
+
+    def __hash__(self) -> int:
+        return hash(self._id)
+
+    def __repr__(self) -> str:
+        return f'<CreId {self._id}>'
+
+    def __str__(self) -> str:
+        return self._id
+
+
+@serializable.serializable_class
+class Requirement:
+    """
+    A requirement comprising a standard.
+    """
+
+    def __init__(
+        self, *,
+        bom_ref: Optional[Union[str, BomRef]] = None,
+        identifier: Optional[str] = None,
+        title: Optional[str] = None,
+        text: Optional[str] = None,
+        descriptions: Optional[Iterable[str]] = None,
+        open_cre: Optional[Iterable[CreId]] = None,
+        parent: Optional[Union[str, BomRef]] = None,
+        properties: Optional[Iterable[Property]] = None,
+        external_references: Optional[Iterable[ExternalReference]] = None,
+    ) -> None:
+        self._bom_ref = bom_ref_from_str(bom_ref)
+        self.identifier = identifier
+        self.title = title
+        self.text = text
+        self.descriptions = descriptions or []  # type:ignore[assignment]
+        self.open_cre = open_cre or []  # type:ignore[assignment]
+        self._parent = bom_ref_from_str(parent)
+        self.properties = properties or []  # type:ignore[assignment]
+        self.external_references = external_references or []  # type:ignore[assignment]
+
+    def __lt__(self, other: Any) -> bool:
+        if isinstance(other, Requirement):
+            return (_ComparableTuple((self.bom_ref, self.identifier))
+                    < _ComparableTuple((other.bom_ref, other.title)))
+        return NotImplemented
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, Requirement):
+            return hash(other) == hash(self)
+        return False
+
+    def __hash__(self) -> int:
+        return hash((
+            self.bom_ref, self.identifier, self.title, self.text, tuple(self.descriptions),
+            tuple(self.open_cre), self.parent, tuple(self.properties), tuple(self.external_references)
+        ))
+
+    def __repr__(self) -> str:
+        return f'<Requirement bom-ref={self.bom_ref}, identifier={self.identifier}, title={self.title}, ' \
+               f'text={self.text}, parent={self.parent}>'
+
+    @property
+    @serializable.json_name('bom-ref')
+    @serializable.type_mapping(BomRefHelper)
+    @serializable.xml_attribute()
+    @serializable.xml_name('bom-ref')
+    def bom_ref(self) -> BomRef:
+        """
+        An optional identifier which can be used to reference the requirement elsewhere in the BOM.
+        Every bom-ref MUST be unique within the BOM.
+
+        If a value was not provided in the constructor, a UUIDv4 will have been assigned.
```

Comment:
> If a value was not provided in the constructor, a UUIDv4 will have been assigned.

has this been tested?

Reply:
Now introduced some additional tests for this in `cyclonedx/model/definition.py`

Reply:
but still, there is not UUIDv4 assigned.

Reply:
just remove this line and you are good.

Reply:
I mean, if you actually intended to have these `bom-ref`s being non-null on rendering, like we do for `component`s, then  you should hook into `BomRefDiscriminator.from_bom()`

Reply:
Ah ok, I see now your point. 

I think assigning a non-required field automatically would cause a lot more problems in the downstream later on. I also do not have a scenario how to reference/use a `bom-ref` of a requirement elsewhere within the BOM. I'll just remove the lines.

Reply:
>  also do not have a scenario how to reference/use a bom-ref of a requirement elsewhere within the BOM.

Here is one: `$.declarations.attestations[].map[].requirement` - <https://cyclonedx.org/docs/1.6/json/#declarations_attestations_items_map_items_requirement>
> The bom-ref to the requirement being attested to.

Reply:
I've updated this comment as you've suggested. Resolving this one.